libdnf/dnf-context.c

```diff
@@ -1420,24 +1419,17 @@ dnf_context_setup_enrollments(DnfContext *context, GError **error)
     if (getuid () != 0)
         return TRUE;
 
-    for (i = 0; cmds[i] != NULL; i++) {
-        int child_argc;
-        g_auto(GStrv) child_argv = NULL;
-        int estatus;
+#ifdef RHSM_SUPPORT
+    g_autoptr(RHSMContext) rhsm_ctx = rhsm_context_new ();
+    g_autofree gchar *repofname = g_build_filename (priv->repo_dir,
+                                                    "redhat.repo",
```

Comment:
`redhat.repo` coming with default container is generated by `sub-man attach`, but then you have to update it. That's point of whole rhsm thing. Because based on entitlements and products you generate different redhat.repo.